helper/schema/field_reader_config.go

```diff
@@ -216,13 +216,65 @@ func (r *ConfigFieldReader) readPrimitive(
 	}, nil
 }
 
+func (r *ConfigFieldReader) readList(
+	address []string, schema *Schema) (FieldReadResult, error) {
+
+	addrPadded := make([]string, len(address)+1)
+	copy(addrPadded, address)
+
+	// Get the number of elements in the list
+	addrPadded[len(addrPadded)-1] = "#"
+	countResult, err := r.readPrimitive(
+		strings.Join(addrPadded, "."), &Schema{Type: TypeInt})
+	if err != nil {
+		return FieldReadResult{}, err
+	}
+	if !countResult.Exists {
+		// No count, means we have no list
+		countResult.Value = 0
+	}
+
+	// If we have an empty list, then return an empty list
+	if countResult.Computed || countResult.Value.(int) == 0 {
+		return FieldReadResult{
+			Value:    []interface{}{},
+			Exists:   countResult.Exists,
+			Computed: countResult.Computed,
+		}, nil
+	}
+
+	// Go through each count, and get the item value out of it
+	result := make([]interface{}, countResult.Value.(int))
+	for i, _ := range result {
+		idx := strconv.Itoa(i)
+		addrPadded[len(addrPadded)-1] = idx
+		rawResult, err := r.readField(addrPadded, true)
+		if err != nil {
+			return FieldReadResult{}, err
+		}
+		if !rawResult.Exists {
+			// This should never happen, because by the time the data
+			// gets to the FieldReaders, all the defaults should be set by
+			// Schema.
+			panic("missing field in list: " + strings.Join(addrPadded, "."))
```

Comment:
Right, that's more or less inherited from the original generic `ReadList` function, but it's still valid here too. 😄